### PR TITLE
found another way to get repo name in gh actions

### PR DESCRIPTION
**Pull request by n3wscott**
also fixed the real vet errors.

### CODE REVIEW

File: .github/workflows/go-pr-verify.yaml [hunk 1]
```diff
@@ -48,20 +48,20 @@ jobs:
     - name: Check out code onto GOPATH
       uses: actions/checkout@v2
       with:
-        fetch-depth: 1
-        path: ./src/knative.dev/${{github.event.pull_request.head.repo.name}}
+        fetch-depth: 14
```
Comment (mattmoor) [inline .github/workflows/go-pr-verify.yaml:51]: Why 14?

Comment (n3wscott) [reply .github/workflows/go-pr-verify.yaml]: 😬

**Pull request merged**